lib/json/JsonGenerator.js

```diff
@@ -164,7 +164,7 @@ class JsonGenerator extends Generator {
 		const jsonStr = stringifySafe(finalJson);
 		const jsonExpr =
 			jsonStr.length > 20 && typeof finalJson === "object"
-				? `JSON.parse(${JSON.stringify(jsonStr)})`
+				? `JSON.parse('${jsonStr}')`
```

Comment:
This need to escape single quotes in the json. And also `\`.

Reply:
Thanks for catching this! I've added escaping to handle those cases.